src/Runner.Common/JobServerQueue.cs

```diff
@@ -819,6 +820,18 @@ private List<TimelineRecord> MergeTimelineRecords(List<TimelineRecord> timelineR
             return mergedRecords;
         }
 
+        private void removeDuplicatedIssues(TimelineRecord timelineRecord)
+        {
+            Dictionary<string, Issue> issuesMap = new Dictionary<string, Issue>();
+            foreach (var i in timelineRecord.Issues)
+            {
```

Comment:
The message text is probably not the right key, here.  What about legitimate duplicates  (i.e. the same error happens several times)?

It'd be nice if `Issue` included a high-precision timestamp or something so that we could distinguish between those.

Reply:
That's a broader question of how to we define the identity of an `Issue`. Message is the most obvious one because that's what an user effectively see, I can add `Type`, `Category` and `IsInfrastructureIssue` to the comparison to include all fields (or just object's HashCode). Timestamp is not ideal because the main problem is that the same Issue is reported in different places. So we still will have duplicated messages, but with distinct timestamps. :D

Reply:
One potentially simple fix might be to change `TimelineRecord::Issues` to be backed by an order-preserving `Set` instead of a `List`.  That would automatically cull duplicates if the same issue reference was added twice.

Reply:
I thought about it but ideally we should fix the real issue, which is in this case: why are duplicates even added? Errors are being logged to the context all over the place and that's the result.
This pr is just a duck tape to fix this particular bug tbh. :D

Reply:
The PR is purely for templating issues though, right? Templating issues have unique messages because of the `/foo/path (Line: 8, Col: 7)` format. I can't think of a legitimate duplicate templating issue that would help the action author here.

Reply:
Regarding removing 'legit' duplicates: issue message contains the line and column numbers, so even if two exceptions are the same they won't be deduplicated because effectively the messages will be different.